cluster/config.go

```diff
@@ -118,5 +121,14 @@ func ConfigProcess() {
 		if err != nil {
 			log.Fatal(4, "CLU Config: swim-bind-addr is not a valid TCP address: %s", err.Error())
 		}
+
+		if swimAdvertiseAddrStr != "" {
+			swimAdvertiseAddr, err = net.ResolveTCPAddr("tcp", swimAdvertiseAddrStr)
+			if err != nil {
+				log.Fatal(4, "CLU Config: swim-advertise-addr is not a valid TCP address: %s", err.Error())
+			}
```

Comment:
technically, this address only needs to resolve on other peers i suppose, but i agree it makes sense to just check it on all peers.

Reply:
The memberlist implementation seems a bit fragile.  Though the docs just specify that AdvertiseAddr is a "string", it actually has to be an IP address and MT will fail to start if the IP cant be parsed by net.ParseIP().

I would have thought it would make sense to have peers use a "hostname" to reach a node, but memberlist does not support that.